app/controllers/devise/devise_authy_controller.rb

```diff
@@ -76,22 +77,28 @@ def POST_enable_authy
 
   # Disable 2FA
   def POST_disable_authy
-    response = Authy::API.delete_user(:id => resource.authy_id)
+    authy_id = resource.authy_id
+    begin
+      resource.update_attributes(authy_enabled: false, authy_id: nil)
 
-    if response.ok?
-      resource.update_attribute(:authy_enabled, false)
-      resource.update_attribute(:authy_id, nil)
-      forget_device
+      # Delete user on Authy if no other users in db are using that authy_id
+      if resource_class.where(authy_id: authy_id).empty?
```

Comment:
It seems like this line and the line under it are the only two unique things that Catalyst needs to do. The rest can be overlayed using a call to `super`